core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -145,17 +149,22 @@ private LocalDate getCurrentDayDate(UserAnonymizedDTO userAnonymized)
 
 	private <T extends IntervalActivity> void addMissingInactivity(Goal activeGoal, ZonedDateTime dateAtStartOfInterval,
 			Set<T> activityEntitiesAtDate, UserAnonymizedDTO userAnonymized,
-			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier)
+			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier, Consumer<T> existingEntityInactivityCompletor)
 	{
-		if (!containsActivityForGoal(activityEntitiesAtDate, activeGoal))
+		Optional<T> activityForGoal = getActivityForGoal(activityEntitiesAtDate, activeGoal);
+		if (activityForGoal.isPresent())
+		{
+			existingEntityInactivityCompletor.accept(activityForGoal.get());
```

Comment:
That's what the method name suggests and what happens on line 161 below, but above here you concluded that there is activity for this goal, so you're adding activity to the existing entity, right? That doesn't sound logical in a method named like this.